conan/tools/google/layout.py

```diff
@@ -1,16 +1,21 @@
 import os
 
 
-def bazel_layout(conanfile, src_folder="."):
+def bazel_layout(conanfile, src_folder=".", build_folder=".", target_folder=None):
     """Bazel layout is so limited. It does not allow to create its special symlinks in other
     folder. See more information in https://bazel.build/remote/output-directories"""
     subproject = conanfile.folders.subproject
     conanfile.folders.source = src_folder if not subproject else os.path.join(subproject, src_folder)
-    conanfile.folders.build = "."  # Bazel always build the whole project in the root folder
-    # FIXME: Keeping backward-compatibility. Defaulting to "conan" in Conan 2.x.
+    # Bazel always build the whole project in the root folder, but consumer can put another one
+    conanfile.folders.build = build_folder if not subproject else os.path.join(subproject, build_folder)
```

Comment:
I am missing the "conan" folder somewhere?

Reply:
Finally, we improved the warning message if it's not defined a generator folder previously.